agent/hcp/telemetry/otel_sink.go

```diff
@@ -85,28 +89,19 @@ func NewOTELSink(opts *OTELSinkOpts) (*OTELSink, error) {
 		return nil, fmt.Errorf("ferror: provide valid context")
 	}
 
-	logger := hclog.FromContext(opts.Ctx).Named("otel_sink")
-
-	filterList, err := newFilterRegex(opts.Filters)
-	if err != nil {
-		logger.Error("Failed to initialize all filters", "error", err)
+	if opts.ConfigProvider == nil {
+		return nil, fmt.Errorf("ferror: provide valid config provider")
 	}
 
-	attrs := make([]attribute.KeyValue, 0, len(opts.Labels))
-	for k, v := range opts.Labels {
-		kv := attribute.KeyValue{
-			Key:   attribute.Key(k),
-			Value: attribute.StringValue(v),
-		}
-		attrs = append(attrs, kv)
-	}
+	logger := hclog.FromContext(opts.Ctx).Named("otel_sink")
+
 	// Setup OTEL Metrics SDK to aggregate, convert and export metrics periodically.
-	res := resource.NewWithAttributes("", attrs...)
```

Comment:
We now use a resource without attributes since labels can change. 

Once created, a resource can't change its attributes. We will now send labels at the metric level instead of OTEL resource level.

Reply:
Okay yeah, understood. Payload size will increase, but there shouldn't be any backend change because of this since we wind up adding the attributes to each sample regardless during the OTLP > Prom conversion

Reply:
Sounds good, let me cut a ticket to look into this and gather info. If it does become a problem, I think we should fast track the compression plan that we had previously, which would mean:
1. Consul/collector should compress export payloads
2. TGW should be able to handle compression (think it currently doesn't?)

Reply:
> TGW should be able to handle compression (think it currently doesn't?)

Correct that it currently doesn't. There's a trade-off where we'll up our CPU/mem usage in Consul + CTGW but for these repetitive sample labels I imagine it's worth it. When we address this it'd be nice to see how the increased cost of compression helps us w/ payload size.

We use `snappy` encoding already to compress payloads in the gateway, and I see that's a dependency of Consul already, that may be one route (adding `Content-Encoding: snappy`)

Reply:
Change from http to grpc would result in a better way of handling payload size as well. We should probably backlog this as the HIP client already uses grpc over traefik and we should probably do the same to reduce the egress charges for customers and for our own hcp clusters.